src/vs/editor/contrib/codeAction/browser/codeAction.ts

```diff
@@ -124,9 +124,8 @@ export async function getCodeActions(
 	const disposables = new DisposableStore();
 	const promises = providers.map(async provider => {
 		try {
-			const codeActionsPromise = Promise.resolve(provider.provideCodeActions(model, rangeOrSelection, codeActionContext, cts.token));
```

Comment:
what was the purpose of the 1.25 second race condition? 

if we want that in, we'd have to change it to something a little higher (maybe 5000 ms), but I think reverting to this should be okay?

Reply:
This is really important to keep because it ensures progress for code actions isn't reported too eager. The race timeout is actually not a 'race condition', it only makes sure the callback is only called when the given promise takes longer than N millis. So, no progress for fast code actions which is the goal here. 

What's the context here and why do you want to change this?

Reply:
I have pushed [8d6b278](https://github.com/microsoft/vscode/pull/232690/commits/8d6b278d7243458c848c1f5fee2fc9d8e0073747) which keeps your fix intact and still delays progress reporting